lib/_emerge/Scheduler.py

```diff
@@ -241,6 +241,25 @@ def __init__(
         for root in self.trees:
             self._config_pool[root] = []
 
+        features = self.settings.features
+        self._jobserver_fd = None
+        self._jobserver_tokens = {}
+        if "jobserver-token" in features:
+            makeflags = self.settings.get("MAKEFLAGS", "").split()
+            jobserver_path = None
+            for flag in makeflags:
+                if flag.startswith("--jobserver-auth="):
+                    flag = flag.removeprefix("--jobserver-auth=")
+                    if flag.startswith("fifo:"):
+                        jobserver_path = flag.removeprefix("fifo:")
+                    else:
+                        # TODO: print a warning?
+                        jobserver_path = None
+            if jobserver_path is not None:
+                # TODO: print a warning otherwise?
+                # TODO: where to close it?
+                self._jobserver_fd = os.open(jobserver_path, os.O_RDWR | os.O_NONBLOCK)
```

Comment:
Should we also implicitly set `_max_jobs` to a sensible value if `FEATURES=jobserver-token` and a `jobserver_path` is found? Otherwise, uses have to explicitly set a max job limit.

Reply:
What's a "sensible value"? We don't want to force `--jobs` when users didn't request it. We don't need to set upper limit because the tokens will generally work that. If people set a lower limit, then they can leave a bunch of tokens free for actual builds, i.e. avoid Portage taking all the tokens and possibly underutilizing them.